### PR TITLE
Vidya

### PR DESCRIPTION
**Pull request by Vidya-Chandrasekhar**
ExceptionsAndLogging pull request

### CODE REVIEW

**Comment by Vidya-Chandrasekhar** [comment]
conflict in README

File: src/main/java/com/zipcodewilmington/phone/PhoneNumberFactory.java [hunk 1]
```diff
@@ -21,14 +21,30 @@ private PhoneNumberFactory() {
      * @return array of randomly generated PhoneNumber objects
      */ //TODO - Implement logic
     public static PhoneNumber[] createRandomPhoneNumberArray(int phoneNumberCount) {
-        return null;
+
+        PhoneNumber[] phoneArray = new PhoneNumber[phoneNumberCount];
+
+        phoneArray[0] = createRandomPhoneNumber();
```
Comment (LeonHunter0) [inline src/main/java/com/zipcodewilmington/phone/PhoneNumberFactory.java]: remove this line

File: src/main/java/com/zipcodewilmington/phone/PhoneNumberFactory.java [hunk 2]
```diff
@@ -21,14 +21,30 @@ private PhoneNumberFactory() {
...
     }
 
+
     /**
      * @return an instance of PhoneNumber with randomly generated phone number value
      */ //TODO - Implement logic
     private static PhoneNumber createRandomPhoneNumber() {
-        return createPhoneNumberSafely(-1, -1, -1);
+
+        RandomNumberFactory randNum = new RandomNumberFactory() {
```
Comment (LeonHunter0) [inline src/main/java/com/zipcodewilmington/phone/PhoneNumberFactory.java]: Clever use of anonymous inner classes... 🤔 
However, the methods in `RandomNumberFactory` are `static`, there is no need for instantiation.
```java
int areaCode = RandomFactory.createInteger(100, 999);
int centralOfficeCode = RandomFactory.createInteger(100, 999);
int phoneLineCode = RandomFactory.createInteger(1000, 9999);

return createPhoneNumberSafely(areaCode, centralOfficeCode, phoneLineCode);
```

File: src/main/java/com/zipcodewilmington/phone/PhoneNumberFactory.java [hunk 3]
```diff
@@ -39,15 +55,30 @@ private static PhoneNumber createRandomPhoneNumber() {
      * @return a new phone number object
      */ //TODO - if input is valid, return respective PhoneNumber object, else return null
     public static PhoneNumber createPhoneNumberSafely(int areaCode, int centralOfficeCode, int phoneLineCode) {
-        return createPhoneNumber(null);
+        try {
+            String phoneNumber = "(" + areaCode + ")" + "-" + centralOfficeCode + "-" + phoneLineCode;
```
Comment (LeonHunter0) [inline src/main/java/com/zipcodewilmington/phone/PhoneNumberFactory.java]:
```java
String phoneNumberValue = String.format("(%s)-%s-%s", areaCode, centralOfficeCode, phoneLineCode);
```

`String.format` performs better than concatenating several strings together.

File: src/main/java/com/zipcodewilmington/phone/PhoneNumberFactory.java [hunk 4]
```diff
@@ -39,15 +55,30 @@ private static PhoneNumber createRandomPhoneNumber() {
...
+            return createPhoneNumber(phoneNumber);
+        } catch (InvalidPhoneNumberFormatException e) {
+
+            logger.info("(" + areaCode + ")" + "-" + centralOfficeCode + "-" + phoneLineCode + " is not a valid phone number ");
```
Comment (LeonHunter0) [inline src/main/java/com/zipcodewilmington/phone/PhoneNumberFactory.java]:
```java
String logMessage = String.format("[ %s ] is not a valid phone number\n", phoneNumberValue);
logger.warn(logMessage);
```
is a bit cleaner.

File: src/main/java/com/zipcodewilmington/phone/PhoneNumberFactory.java [hunk 5]
```diff
@@ -39,15 +55,30 @@ private static PhoneNumber createRandomPhoneNumber() {
...
      * @throws InvalidPhoneNumberFormatException - thrown if phoneNumberString does not match acceptable format
      */ // TODO - Add throws statement to method signature
-    public static PhoneNumber createPhoneNumber(String phoneNumberString) {
-        return null;
+    public static PhoneNumber createPhoneNumber(String phoneNumberString) throws InvalidPhoneNumberFormatException {
+
+
+        PhoneNumber phone = new PhoneNumber(phoneNumberString);
+
+        logger.info("Attempting to create a new PhoneNumber object with a value of " + "(" + phone.getAreaCode() + ")" + "-" + phone.getCentralOfficeCode() + "-" + phone.getPhoneLineCode());
```
Comment (LeonHunter0) [inline src/main/java/com/zipcodewilmington/phone/PhoneNumberFactory.java]: You are logging an event prior to its occurrence!

Additionally, something like...
```java
String logMessage = String.format("Attempting to create a new PhoneNumber object with a value of ` %s `", phoneNumberString);
logger.info(logMessage);
```
is a bit easier to read.

**Pull request closed without merging**